### PR TITLE
DHFPROD-3313:Keep xml/xslt when reinstallUserModules() is called

### PR DESCRIPTION
**Pull request by srinathgit**
Keep xml/xslt when reinstallUserModules() is called

### CODE REVIEW

File: web/src/main/java/com/marklogic/hub/web/service/DataHubService.java [hunk 1]
```diff
@@ -190,7 +190,6 @@ private void installUserModules(HubConfig hubConfig, boolean forceLoad, DeployUs
         loadHubArtifactsCommand.setHubConfig(hubConfig);
         loadHubArtifactsCommand.setForceLoad(forceLoad);
 
-        commands.add(generateFunctionMetadataCommand);
```
Comment (rjrudin) [inline web/src/main/java/com/marklogic/hub/web/service/DataHubService.java]: Isn't this still needed for when e.g. a user's project is first installed?

File: marklogic-data-hub/src/main/java/com/marklogic/hub/impl/DataHubImpl.java [hunk 2]
```diff
@@ -376,7 +376,8 @@ public void clearUserModules() {
             );
 
             String query =
-                "cts:uris((),(),cts:not-query(cts:collection-query('hub-core-module')))[\n" +
+                "cts:uris((),(),cts:not-query(cts:collection-query(('hub-core-module','http://marklogic.com/entity-services/mapping'" +
```
Comment (rjrudin) [inline marklogic-data-hub/src/main/java/com/marklogic/hub/impl/DataHubImpl.java]: This doesn't really seem safe... I think if a user runs "mlClearModulesDatabase", the user would expect the xml/xslt files generated for their own custom functions to be deleted as well. 

This seems to be addressing a timing issue in QS, right? Where between the time it takes for QS to save the mapping and then reinstall modules, the user may choose to run their mapping? 

In which case - isn't their always a chance that the user's custom mapping functions are deleted and not yet reloaded by the time the user chooses to run the mapping?

I'll play around with this locally. My hunch is that in the QS context, we don't need to clear the modules database in the first place. Clearing modules makes sense when you e.g. start your day and pull from git and you want to ensure any deletions in git are made in your modules database as well. I don't think it's necessary to do while in QS though.

**Comment by rjrudin** [comment]
Take a look at https://github.com/marklogic/marklogic-data-hub/pull/3094 - I think the better fix is to not call "reinstallUserModules" when saving entities/flows/mappings but rather to just call "installUserModules".

I am thinking the call to "reinstall" is at the heart of these intermittent problems people see with modules/mappings disappearing.

**Comment by srinathgit** [comment]
Closing this as #3094 has fix for it.

**Pull request closed without merging**